app/controllers/concerns/authentication.rb

```diff
@@ -18,6 +18,7 @@ def require_authentication
 
     return if current_user?
 
+    session[:course_id] = params[:course_id] if params.key?(:course_id)
```

Comment:
I'd suggest renaming to `last_course_id` to be super clear.

Also, we'll need to update the privacy policy to mention that we're storing this (in an encrypted cookie, as part of the user's session).